### DISCUSSION TITLE
How to pass in data test id for the ImageSlide

### DISCUSSION THREAD

**Discussion by npatel-qb** [igordanchenko/yet-another-react-lightbox#298] [Q&A] [answered]
I have an image slide that I am using as a custom slide for my lightbox. But I can't pass in the data test id down to the Image Slide. Is there any way to achieve that? I did try to wrap the ImageSlide in a div and use that for the data test id but JestDOM only renders the image tags and not the outer div elements. 

<img width="581" alt="Screenshot 2024-08-06 at 9 51 53 AM" src="https://github.com/user-attachments/assets/09c3fde4-c1e5-4df2-9956-0bf505491e6a">

**Comment by igordanchenko** [comment]
Have you tried `imageProps`?

**Reply by npatel-qb** [reply]
Like passing the data test id through the imageProps?

**Reply by igordanchenko** [reply, marked as answer]
Correct. The `imageProps` prop is getting passed through to the underlying `img` tag.

**Reply by npatel-qb** [reply]
Right that makes sense. I couldn't really find a `testID` or `dataTestID` field in the `imageProps`. Is that not supported?

**Reply by igordanchenko** [reply]
`React.ComponentProps` doesn't include `data-` attributes for some reason, so you can either `@ts-ignore` it or add the following module augmentation: 

```ts
declare module "react" {
  interface HTMLAttributes<T> extends AriaAttributes, DOMAttributes<T> {
    [dataAttribute: `data-${string}`]: unknown;
  }
}
```

**Reply by npatel-qb** [reply]
Great thanks a lot for the help!
